### PR TITLE
Feature5 fixes

### PR DESCRIPTION
**Pull request by lingaoliu105**
managed to run the tests and added some bug fixes and refactoring to previous code.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -68,6 +70,7 @@ group :development, :test do
   # See https://guides.rubyonrails.org/debugging_rails_applications.html#debugging-with-the-debug-gem
   # gem "debug", platforms: %i[ mri mingw x64_mingw ]
   gem 'debug'
+  # gem 'ffi'
```
Comment (ExtremelySunnyYK) [inline Gemfile:73]: removed ffi for us

File: app/models/loyalty_program_datum.rb [hunk 2]
```diff
@@ -8,7 +8,7 @@ class LoyaltyProgramDatum < ApplicationRecord
   # def update_amount
   #     # Add amount based on loyalty program id
 
-  #     new_amount = Transactions.where(loyalty_program_id: loyalty_program_id).sum(:amount)
+  #     new_amount = Transactions.where(loya"#{time.year}#{"%02d" % time.month}#{"%02d" % time.day}"lty_program_id: loyalty_program_id).sum(:amount)
```
Comment (ExtremelySunnyYK) [inline app/models/loyalty_program_datum.rb:11]: should we remove this function

**Review by ExtremelySunnyYK — APPROVED** [review]

**Comment by ExtremelySunnyYK** [comment]
No brreaking changes to me, but the don't really understand the flow of the code

File: app/utils/accrual_processor.rb [hunk 3]
```diff
@@ -1,32 +1,64 @@
 class AccrualProcessor < Rails::Application
-  @@current_index = 1
+  # @@current_index = 1
+  @@current_indices = Hash.new
```
Comment (ExtremelySunnyYK) [inline app/utils/accrual_processor.rb:3]: why do we create a hash for current_indices?

File: app/models/loyalty_program_datum.rb (hunk 2 above)
Comment (suvasan19) [reply app/models/loyalty_program_datum.rb:11]: I think we should

File: spec/factories/loyalty_program_datum.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# FactoryBot.define do
```
Comment (suvasan19) [inline spec/factories/loyalty_program_datum.rb:1]: do we needed these if they are commented out?

File: spec/factories/transaction.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# FactoryBot.define do
```
Comment (suvasan19) [inline spec/factories/transaction.rb:1]: Same here, do we needed these if they are commented out?

**Pull request merged**